wordpress-mobile/WordPress-Android
# My Site Dashboard [Phase 2]: Connect Dashboard Cards Store with Source

Parent: #15498

This PR removes the mocked posts data and replaces that with the actual post data, which comes from the newly added `CardsStore` functionality. In effect, this PR connects the `CardsStore` with the `CardsSource` and makes sure that whenever this source is build, or refreshed for that matter, the data gets updated and shown on screen accordingly.

In more detail, this PR:
- Updates the naming convention and packaging structure on all things dashboard cards related (cards `source`, `update`, etc).
- Replaces the mocked posts data, and json utils, with the `CardsResult`, thus removing the need for the mocked posts data related json files altogether.
- Implements the new `CardsSource` logic, based on the `CardsStore`, getting the data from the database, through `Room`, utilizing flow, and fetching the data from the network, including posting the error state if necessary.

-----

To test:
- There is actually nothing to test in this PR, just review the `CardsSource` class and related changes and make sure that the accompanying tests are working as expected.
- When the [WordPress-FluxC-Android#2197](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2197) will get merged, and then this PR gets updated with the new version of FluxC, that which contains the update, then loading the dashboard feed for posts can be tested as a whole.

-----

To manual test:
- If you really want to test the solution, and I suggest you should:
  - Uncomment the `localFluxCPath = "../WordPress-FluxC-Android"` from your `local-builds.gradle` to work with you local `FluxC` repo.
  - Find the `issue/dashboard-cards-db` branch in `FluxC` and check it out.
  - Now build the project as usual and verify that
    - When the `MySiteDashboardPhase2FeatureConfig` is off, then the dashboard feed for posts doesn't appears on the `My Site` tab.
    - When the `MySiteDashboardPhase2FeatureConfig` is on, then the dashboard feed for posts does appears on the `My Site` tab, and with real, non-mocked data.
- However, mind the fact that you would also need to:
  - Either apply the [My Site Dashboard: Cards Endpoint](https://code.a8c.com/D69742) differential on your sandbox, proxy your device on it. Let me know if you need any help with that!
  - Or, you can wait for the [My Site Dashboard: Cards Endpoint](https://code.a8c.com/D69742) differential to be deployed. Then, you would be able to test the solution as usual.

-----

Merge instructions:
- [x] ~~Wait for the new API endpoint to be deployed (see [D69742](https://code.a8c.com/D69742)).~~
- [x] Wait for the accompanying [WordPress-FluxC-Android#2197](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2197) PR to be merged.
- [x] Update `fluxcVersion` with the once that includes the above FluxC solution (with either the `develop` hash or pointing to the newest release). PS: This PR will fail until this step is complete.
- [x] Remove the `[DO NOT MERGE]` from the title.
- [x] Remove the `[Status] Not Ready for Merge` label.

-----

## Regression Notes
1. Potential unintended areas of impact

When the `Phase 2` feature flag is:
- Off, then N/A.
- On, then the `My Site` tab and its inner cards and items might somehow be affected by the loading of the newly added `posts` dashboard cards.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` and `To manual test` sections above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
